client/src/App.tsx

```diff
@@ -1,38 +1,195 @@
+
+//If the player !isHuman there is a setTimeout to mimic a human choosing a stat
+//chooseRandomStat selects a random stat from the current player's card
+//playRound then runs with the chosen stat
+
+//If the player isHuman there should be a prompt to pick a stat from the card i.e. "Please chose a stat player 2"
+//The player clicks on a stat which triggers playRound to run with their chosen stat
+
+//showAllCards reveals all of the player's top cards
+//A setTimeout is used to allow time for all the cards to be shown
+//calculateRoundWinner then compares the value of chosen stat with the value of that stat on the top card of every other player
+//The player with the card with the highest value for that characteristic wins the round
+//If there is a draw between multiple highest values then the current player wins the round
+
+//endRound then runs to end the round and prepare for the next round
+//hideAllCards turns all the cards face down
+//A setTimeout is used to allow time for all of the cards to be hidden
+//The top card, i.e. the card used this round, for each player is set to the bottom of their hand
+
+//If the currentRoundRef === game.totalRounds then the game ends and a winner is declared
+//If the curerntRoundRef !== game.totalRounds then startRound() runs and to start the round for the next player
+
 function App() {
   const [game, setGame] = useState<Game>(
     initialiseGame(DEFAULT_PLAYERS, DEFAULT_ROUNDS)
   );
   const player1 = game.players[0];
   const player2 = game.players[1];
 
+  const { players, totalRounds, roundWinners } = game;
+
+  const currentRoundRef = useRef<number>(0);
+  const currentPlayerRef = useRef<Player>(players[0]);
+
+  const showPlayerCard = (player: Player) => {
+    setGame((prevGame: Game) => {
+      return {
+        ...prevGame,
+        players: updatePlayerIsCardShown(prevGame.players, player.id),
+      };
+    });
+  };
+
+  const showAllCards = () => {
+    setGame((prevGame: Game) => {
+      return {
+        ...prevGame,
+        players: updatePlayerIsCardShownAll(prevGame.players, true),
+      };
+    });
+  };
+
+  const hideAllCards = () => {
+    setGame((prevGame: Game) => {
+      return {
+        ...prevGame,
+        players: updatePlayerIsCardShownAll(prevGame.players, false),
+      };
+    });
+  };
+
+  const nextPlayer = () => {
+    const prevPlayer = currentPlayerRef.current;
+    const prevPlayerIndex = players.findIndex(
+      (player) => player.id === prevPlayer.id
+    );
+    let nextPlayer: Player;
+    if (prevPlayerIndex === players.length - 1) {
+      nextPlayer = players[0];
+    } else {
+      nextPlayer = players[prevPlayerIndex + 1];
+    }
+    currentPlayerRef.current = nextPlayer;
+    startRound(nextPlayer);
+  };
+
+  const startRound = (player: Player) => {
+    currentRoundRef.current = currentRoundRef.current + 1;
+
+    setGame((prevGame: Game) => {
+      return {
+        ...prevGame,
+        currentRound: prevGame.currentRound + 1,
```

Comment:
Here I think if we got rid of currentRound in Game then we don't need to calculate it.

Reply:
So, the currentRoundRef was necessary because currentRound in state didn't work to tell the game to stop playing. See comment above.

I was going to remove currentRound from state and just have it in ref, but we need it in state so that we update the RoundTracker to show the current round at the start of each new round.